packages/pages/src/vite-plugin/build/buildStart/rendering/wrapper.ts

```diff
@@ -64,12 +63,12 @@ const getCssTags = (
   const [file, info] = entry;
 
   seen.add(file);
-  const cssFiles = info.css || [];
+  const cssFiles = new Set(info.css) || [];
   (info.imports || [])
     .flatMap((f) => getCssTags(f, manifest, seen))
-    .forEach((f) => cssFiles.push(f));
+    .forEach((f) => cssFiles.add(f));
 
-  return cssFiles;
+  return Array.from(cssFiles);
```

Comment:
just continue to return a set and do a `cssFiles.forEach` to append them to the head.